runtime/syntax/sed.vim

```diff
@@ -19,6 +19,10 @@ syn region sedAddress   matchgroup=Special start="[{,;]\s*/\(\\/\)\="lc=1 skip="
 syn region sedAddress   matchgroup=Special start="^\s*/\(\\/\)\=" skip="[^\\]\(\\\\\)*\\/" end="/I\=" contains=sedTab,sedRegexpMeta
 syn match sedComment	"^\s*#.*$"
 syn match sedFunction	"[dDgGhHlnNpPqQx=]\s*\($\|;\)" contains=sedSemicolon,sedWhitespace
+if !exists("g:sed_syntax_style") || g:sed_syntax_style !=? "bsd"
+    syn match sedComment	"\s*#.*$"
```

Comment:
This pattern doesn't need the leading whitespace as it only currently follows `sedFunction` which itself includes trailing whitespace and consumes it all.

This doesn't make any difference at the moment but might in the future because this syntax file explicitly matches whitespace with `sedWhitespace`.

This is also taking precedence over the full line comment, on line 20, which now never matches.  Maybe someone has a reason to distinguish between them?  If so, you could name the new one `sedTailComment` and if not you could just move the full line definition to after the conditional block.

Reply:
I hadn't thought of distinguishing `sedLineComment` and `sedTailComment`, but sure, maybe it helps. But I don't understand the logic enough to make it work. Something with `contains` or `nextgroup` to make sure it comes after one of the other groups?